tests/integration/test_kalman_filter_with_sparse_sites.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import tensorflow as tf
+import pytest
+from gpflow.config import default_float
+
+from markovflow.kernels.matern import Matern12
+from markovflow.mean_function import LinearMeanFunction
+from markovflow.models.gaussian_process_regression import GaussianProcessRegression
+from markovflow.kalman_filter import KalmanFilterWithSparseSites, UnivariateGaussianSitesNat
+from markovflow.likelihoods import MultivariateGaussian
+
+@pytest.fixture(
+    name="time_step_homogeneous", params=[(0.01, True), (0.01, False), (0.001, True), (0.001, False)],
+)
+def _time_step_homogeneous_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="kalman_gpr_setup")
+def _setup(batch_shape, time_step_homogeneous):
+    """
+    Create a Gaussian Process model and an equivalent kalman filter model
+    with more latent states than observations. 
+    FIXME: Currently batch_shape isn't used.
+    """
+    dt, homogeneous = time_step_homogeneous
+
+    time_grid = np.arange(0.0, 1.0, dt)
+    if not homogeneous:
+        time_grid = np.sort(np.random.choice(time_grid, 50, replace=False))
+
+    time_points = time_grid[::10]
+    observations = np.sin(12 * time_points[..., None]) + np.random.randn(len(time_points), 1) * 0.1
+
+    input_data = (
+        tf.constant(time_points, dtype=default_float()),
+        tf.constant(observations, dtype=default_float()),
+    )
+
+    observation_covariance = 1.0  # Same as GPFlow default
+    kernel = Matern12(lengthscale=1.0, variance=1.0, output_dim=observations.shape[-1])
+    kernel.set_state_mean(tf.random.normal((1,), dtype=default_float()))
+    gpr_model = GaussianProcessRegression(
+            input_data=input_data,
+            kernel=kernel,
+            mean_function=LinearMeanFunction(1.1),
+            chol_obs_covariance=tf.constant([[np.sqrt(observation_covariance)]], dtype=default_float()),
+        )
+
+    prior_ssm = kernel.state_space_model(time_grid)
+    emission_model = kernel.generate_emission_model(time_grid)
+    observations_index = tf.where(tf.equal(time_grid[..., None], time_points))[:, 0][..., None]
+
+    observations -= gpr_model.mean_function(time_points)
+
+    nat1 = observations / observation_covariance
+    nat1 = tf.scatter_nd(observations_index, nat1, time_grid[..., None].shape)
+    nat2 = (-0.5 / observation_covariance) * tf.ones_like(nat1)[..., None]
+    lognorm = tf.zeros_like(nat1)
+    sites = UnivariateGaussianSitesNat(nat1=nat1, nat2=nat2, log_norm=lognorm)
```

Comment:
I m not sure why you prefer dense sites for this method 
For training purposes that will create bigger parameters with no gradient flowing through these.

I guess I had in mind to then refactor the SDESSM class you have to use the Sparse CVI so maybe the dense sites make more sense.

Reply:
I see what you mean. So, posterior is only dependent on the data-sites so it can be done this way. 

My assumption was that all-sites should also contribute to the posterior which it doesn't. Shouldn't it also play a role? Otherwise, during inference what are the all-sites used for?

Reply:
the assumption of this model is that the all sites are forever 0